cli/running/watchdog.go

```diff
@@ -146,6 +157,27 @@ func (wd *Watchdog) Start() error {
 	return nil
 }
 
+// startIntegrityChecks launches gorountine that performs periodic integrity checks.
+func (wd *Watchdog) startIntegrityChecks() {
+	ticker := time.NewTicker(wd.integrityCheckPeriod)
+
+	go func() {
+		for {
```

Comment:
I don't see a normal path exit condition in the loop.

Reply:
I've added an exit upon `wd.done`